Windows IntelliJ to WSL Project: Project Scan, Symlink and Execution Issues

**Environment**:
- *Windows IntelliJ*: IntelliJ IDEA 2025.2.3 (Ultimate Edition)
- *SAP Commerce Developer Toolset*: 2025.2.4.3
- *WSL*: 2.6.1.0
- *WSL Distro*: Ubuntu 24.04.3 LTS

IntelliJ Support for opening projects placed on a WSL distro has improved a lot and general file search and indexing usually works without problems.
However, we noticed some problems when using the plugin in this context, mostly related to project creation:
- Initial scan for opening a project takes a long time.
- Symlinks are not resolved
- JDKs are not discovered
- Trying to run the project via "Local SAP CX Server" throws an error:
```
Error: Could not find or load main class de.hybris.bootstrap.codegenerator.CodeGenerator
Caused by: java.lang.ClassNotFoundException: de.hybris.bootstrap.codegenerator.CodeGenerator
```

Opening a project initially created using Gateway (basically from an IntelliJ on WSL itself) has the proper folder structure until it I try to refresh, but starting the server does not work either way.

It would be awesome if the support could be added.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,7 +1,8 @@
 ## [2025.3.1]
 
 <cite>Release contributors</cite>
-- 13 PR(s) by [Mykhailo Lytvyn](https://github.com/epam/sap-commerce-intellij-idea-plugin/pulls?q=milestone%3A2025.3.1+author%3Amlytvyn+is%3Apr)
+
+- 14 PR(s) by [Mykhailo Lytvyn](https://github.com/epam/sap-commerce-intellij-idea-plugin/pulls?q=milestone%3A2025.3.1+author%3Amlytvyn+is%3Apr)
 - 4 PR(s) by [Eugeni Kalenchuk](https://github.com/epam/sap-commerce-intellij-idea-plugin/pulls?q=milestone%3A2025.3.1+author%3Aekalenchuk+is%3Apr++is%3Apr)
 - 1 PR by [Maksym Shyshkin](https://github.com/epam/sap-commerce-intellij-idea-plugin/pulls?q=milestone%3A2025.3.1+author%3ASomewhereInBetween+is%3Apr+)
 
@@ -34,6 +35,8 @@ Due severe API changes it is highly advised to backup the project and import it 
 
 ### Other
 - Updated libraries [#1714](https://github.com/epam/sap-commerce-intellij-idea-plugin/pull/1714)
+- New flag to not re-generate code before
+  `Local SAP CX Server` run configuration [#1716](https://github.com/epam/sap-commerce-intellij-idea-plugin/pull/1716)
 
 ## [2025.3.0]
 

--- a/modules/project/compile/src/sap/commerce/toolset/project/compile/GenerateCodeCompilerTask.kt
+++ b/modules/project/compile/src/sap/commerce/toolset/project/compile/GenerateCodeCompilerTask.kt
@@ -28,6 +28,7 @@ import com.intellij.openapi.projectRoots.JavaSdkType
 import com.intellij.openapi.roots.ModuleRootManager
 import com.intellij.util.application
 import com.intellij.util.asSafely
+import sap.commerce.toolset.HybrisConstants
 import sap.commerce.toolset.extensioninfo.EiConstants
 import sap.commerce.toolset.isHybrisProject
 import sap.commerce.toolset.project.compile.context.CompileTaskContext
@@ -54,8 +55,8 @@ class GenerateCodeCompilerTask : CompileTask {
 
         val typeId = context.compileScope.getUserData(CompilerManager.RUN_CONFIGURATION_TYPE_ID_KEY)
         // do not rebuild sources in case of JUnit
-        // see JUnitConfigurationType
-        if ("JUnit" == typeId && !settings.generateCodeOnJUnitRunConfiguration) return true
+        if (HybrisConstants.RunConfiguration.JUNIT == typeId && !settings.generateCodeOnJUnitRunConfiguration) return true
+        if (HybrisConstants.RunConfiguration.SAP_CX == typeId && !settings.generateCodeOnServerRunConfiguration) return true
 
         val preCompileTask = getPreCompileTask(context) ?: return true
 

--- a/modules/project/compile/src/sap/commerce/toolset/project/compile/ProjectAfterCompilerTask.kt
+++ b/modules/project/compile/src/sap/commerce/toolset/project/compile/ProjectAfterCompilerTask.kt
@@ -22,6 +22,7 @@ import com.intellij.openapi.compiler.CompileTask
 import com.intellij.openapi.compiler.CompilerManager
 import com.intellij.openapi.module.Module
 import com.intellij.util.application
+import sap.commerce.toolset.HybrisConstants
 import sap.commerce.toolset.extensioninfo.EiConstants
 import sap.commerce.toolset.isHybrisProject
 import sap.commerce.toolset.project.ProjectConstants
@@ -39,8 +40,8 @@ class ProjectAfterCompilerTask : CompileTask {
 
         val typeId = context.compileScope.getUserData(CompilerManager.RUN_CONFIGURATION_TYPE_ID_KEY)
         // do not rebuild sources in case of JUnit
-        // see JUnitConfigurationType
-        if ("JUnit" == typeId && !settings.generateCodeOnJUnitRunConfiguration) return@runReadAction true
+        if (HybrisConstants.RunConfiguration.JUNIT == typeId && !settings.generateCodeOnJUnitRunConfiguration) return@runReadAction true
+        if (HybrisConstants.RunConfiguration.SAP_CX == typeId && !settings.generateCodeOnServerRunConfiguration) return@runReadAction true
 
         val modules = application.runReadAction<Array<Module>> { context.compileScope.affectedModules }
         val moduleMapping = settings.module2extensionMapping

--- a/modules/project/core/src/sap/commerce/toolset/project/settings/ProjectSettings.kt
+++ b/modules/project/core/src/sap/commerce/toolset/project/settings/ProjectSettings.kt
@@ -1,6 +1,6 @@
 /*
  * This file is part of "SAP Commerce Developers Toolset" plugin for IntelliJ IDEA.
- * Copyright (C) 2019-2025 EPAM Systems <hybrisideaplugin@epam.com> and contributors
+ * Copyright (C) 2019-2026 EPAM Systems <hybrisideaplugin@epam.com> and contributors
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -133,6 +133,11 @@ class ProjectSettings : SerializablePersistentStateComponent<ProjectSettingsStat
         get() = state.generateCodeOnJUnitRunConfiguration
         set(value) {
             updateState { it.copy(generateCodeOnJUnitRunConfiguration = value) }
+        }
+    var generateCodeOnServerRunConfiguration
+        get() = state.generateCodeOnServerRunConfiguration
+        set(value) {
+            updateState { it.copy(generateCodeOnServerRunConfiguration = value) }
         }
     var generateCodeTimeoutSeconds
         get() = state.generateCodeTimeoutSeconds

--- a/modules/project/core/src/sap/commerce/toolset/project/settings/state/ProjectSettingsState.kt
+++ b/modules/project/core/src/sap/commerce/toolset/project/settings/state/ProjectSettingsState.kt
@@ -1,6 +1,6 @@
 /*
  * This file is part of "SAP Commerce Developers Toolset" plugin for IntelliJ IDEA.
- * Copyright (C) 2019-2025 EPAM Systems <hybrisideaplugin@epam.com> and contributors
+ * Copyright (C) 2019-2026 EPAM Systems <hybrisideaplugin@epam.com> and contributors
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -36,6 +36,7 @@ data class ProjectSettingsState(
     @JvmField @OptionTag val ideModulesFilesDirectory: String? = null,
     @JvmField @OptionTag val generateCodeOnRebuild: Boolean = true,
     @JvmField @OptionTag val generateCodeOnJUnitRunConfiguration: Boolean = false,
+    @JvmField @OptionTag val generateCodeOnServerRunConfiguration: Boolean = false,
     @JvmField @OptionTag val generateCodeTimeoutSeconds: Int = 60,
     @JvmField @OptionTag val importOotbModulesInReadOnlyMode: Boolean = true,
     @JvmField @OptionTag val importCustomAntBuildFiles: Boolean = false,

--- a/modules/project/exec/src/sap/commerce/toolset/project/runConfigurations/LocalSapCXConfigurationType.kt
+++ b/modules/project/exec/src/sap/commerce/toolset/project/runConfigurations/LocalSapCXConfigurationType.kt
@@ -1,6 +1,6 @@
 /*
  * This file is part of "SAP Commerce Developers Toolset" plugin for IntelliJ IDEA.
- * Copyright (C) 2019-2025 EPAM Systems <hybrisideaplugin@epam.com> and contributors
+ * Copyright (C) 2019-2026 EPAM Systems <hybrisideaplugin@epam.com> and contributors
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -22,10 +22,11 @@ import com.intellij.execution.configurations.RunConfiguration
 import com.intellij.execution.configurations.SimpleConfigurationType
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.util.NotNullLazyValue
+import sap.commerce.toolset.HybrisConstants
 import sap.commerce.toolset.HybrisIcons
 
 class LocalSapCXConfigurationType : SimpleConfigurationType(
-    "SAPCX",
+    HybrisConstants.RunConfiguration.SAP_CX,
     "SAP CX Start",
     "Start local SAP CX Platform Configuration Type",
     NotNullLazyValue.createValue { HybrisIcons.Y.LOGO_BLUE }

--- a/modules/project/ui/src/sap/commerce/toolset/project/options/ProjectSettingsConfigurableProvider.kt
+++ b/modules/project/ui/src/sap/commerce/toolset/project/options/ProjectSettingsConfigurableProvider.kt
@@ -1,6 +1,6 @@
 /*
  * This file is part of "SAP Commerce Developers Toolset" plugin for IntelliJ IDEA.
- * Copyright (C) 2019-2025 EPAM Systems <hybrisideaplugin@epam.com> and contributors
+ * Copyright (C) 2019-2026 EPAM Systems <hybrisideaplugin@epam.com> and contributors
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -98,6 +98,11 @@ class ProjectSettingsConfigurableProvider(private val project: Project) : Config
                 row {
                     checkBox("Generate code before execution of the JUnit Run Configuration")
                         .bindSelected(projectSettings::generateCodeOnJUnitRunConfiguration)
+                        .enabledIf(generateCodeOnRebuildCheckBox.selected)
+                }
+                row {
+                    checkBox("Generate code before execution of the SAP CX Server Run Configuration")
+                        .bindSelected(projectSettings::generateCodeOnServerRunConfiguration)
                         .enabledIf(generateCodeOnRebuildCheckBox.selected)
                 }
                 row("Code generation timeout:") {

--- a/modules/shared/core/src/sap/commerce/toolset/HybrisConstants.kt
+++ b/modules/shared/core/src/sap/commerce/toolset/HybrisConstants.kt
@@ -1,6 +1,6 @@
 /*
  * This file is part of "SAP Commerce Developers Toolset" plugin for IntelliJ IDEA.
- * Copyright (C) 2019-2025 EPAM Systems <hybrisideaplugin@epam.com> and contributors
+ * Copyright (C) 2019-2026 EPAM Systems <hybrisideaplugin@epam.com> and contributors
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -302,5 +302,11 @@ object HybrisConstants {
     object Folding {
         const val NO_VALUE = "🚫" // ? or 🚫 ∅
         const val DESCRIPTION_PREFIX = "ⓘ"
+    }
+
+    object RunConfiguration {
+        // see JUnitConfigurationType
+        const val JUNIT = "JUnit"
+        const val SAP_CX = "SAPCX"
     }
 }

